tests/spec/puppeteer-spec.js

```diff
@@ -147,4 +147,24 @@ describe("Puppeteer basic tests for the Python Editor.", function() {
         expect(codeName).toEqual("main");
     });
 
+    it("Saves a python file with the correct filename", async function(){
+        const page = await global.browser.newPage();
+        await page._client.send('Page.setDownloadBehavior', {behavior: 'allow', downloadPath: './spec/test-files/'});
+        await page.goto("http://localhost:5000/editor.html");
+        await page.evaluate( () => document.getElementById("script-name").value = "program test")
+        await page.waitFor(500);
```

Comment:
If the page is waiting for something to change, could you do wait loops like the ones in the other tests?
```
for (let ms=0; ms<100; ms++) {
    if (<condition meet>) break;
    await page.waitFor(10);
}
```
Otherwise the tests keep taking more and more time, and they are already very slow.

This applies to all the other `waitFor()`.